publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java

```diff
@@ -1699,6 +1700,52 @@ void shouldNotOverrideUploadDetailsWhenFileIsUpdated() throws BadRequestExceptio
         assertThat(updatedFile.getUploadDetails().getUploadedBy().getValue(), is(not(equalTo(contributorName))));
     }
 
+    @Test
+    void publicationOwnerShouldBeAbleToUpdateLicenseOfPublishedFile() throws ApiGatewayException, IOException {
+        var publication = TicketTestUtils.createPersistedPublicationWithPublishedFiles(customerId, PUBLISHED,
+                                                                                       resourceService);
+        var publishedFile = publication.getAssociatedArtifacts().stream()
+                              .filter(PublishedFile.class::isInstance)
+                              .map(PublishedFile.class::cast)
+                              .findFirst().orElseThrow();
+        var updatedFile = publishedFile.copy().withLicense(randomUri()).buildPublishedFile();
+        var files = publication.getAssociatedArtifacts();
+        files.remove(publishedFile);
+        files.add(updatedFile);
+        var event = ownerUpdatesOwnPublication(publication.getIdentifier(), publication);
+
+        updatePublicationHandler.handleRequest(event, output, context);
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, Publication.class);
+        var updatedPublication = resourceService.getPublication(publication);
+
+        assertThat(updatedPublication.getAssociatedArtifacts().getFirst(), is(equalTo(updatedFile)));
+        assertThat(gatewayResponse.getStatusCode(), is(equalTo(HTTP_OK)));
+    }
+
+    @Test
+    void publicationOwnerShouldNotBeAbleToUpdateOtherFieldsOfPublishedFileThanLicense()
+        throws ApiGatewayException, IOException {
+        var publication = TicketTestUtils.createPersistedPublicationWithPublishedFiles(customerId, PUBLISHED,
+                                                                                       resourceService);
+        var publishedFile = publication.getAssociatedArtifacts().stream()
+                                .filter(PublishedFile.class::isInstance)
+                                .map(PublishedFile.class::cast)
+                                .findFirst().orElseThrow();
+        var updatedFile = publishedFile.copy()
+                              .withLicense(randomUri())
+                              .withName(randomString())
+                              .buildPublishedFile();
+        var files = publication.getAssociatedArtifacts();
+        files.remove(publishedFile);
+        files.add(updatedFile);
+        var event = ownerUpdatesOwnPublication(publication.getIdentifier(), publication);
+
+        updatePublicationHandler.handleRequest(event, output, context);
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(gatewayResponse.getStatusCode(), is(equalTo(HTTP_FORBIDDEN)));
+    }
+
```

Comment:
Suggest adding a test that guards against mismatch between equalsExceptLicense and normal equals if you manage to do this